File: QytetetRuby/pkg/QytetetRuby-0.0.1/lib/dado.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require "singleton"

module ModeloQytetet

  class Dado

    include Singleton

    def initialize

    end

    
    def tirar()    
      rand(6) + 1
    end
    
    
    def to_s()
      
    end
    
  end

end

File: QytetetJava/src/modeloqytetet/Dado.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author jose26398
 */
public class Dado {
    
    int tirar(){    
        return (int)((Math.random() * 6)+1);
    }
    
    private Dado() {
    }
    
    public static Dado getInstance() {
        return DadoHolder.INSTANCE;
    }
    
    private static class DadoHolder {

        private static final Dado INSTANCE = new Dado();
    }
}
